EIPS/eip-1620.md

```diff
@@ -1,296 +1,185 @@
-
-- `stream`
-    - `sender`: the `address` of the entity funding the stream
-    - `recipient`: the `address` where the money is being delivered to
-    - `tokenAddress`: the `address` of the ERC20 token used as payment asset
-    - `balance`: the total funds left in the stream
-    - `timeframe`: as defined below
-    - `rate`: as defined below
-
-```solidity
-  struct Stream {
-    address sender;
+The structure of a stream should be as follows:
+
+- `Stream`
+    - `deposit`: the amount of money to be streamed
+    - `ratePerSecond`: the number of tokens allocated each second to the recipient
+    - `remainingBalance`: the left in the stream
+    - `startTime`: the unix timestamp for when the stream starts
+    - `stopTime`: the unix timestamp for when the stream stops
+    - `recipient`: the address towards which the money is streamed
+    - `sender`: the address of the party funding the stream
+    - `tokenAddress`: the ERC20 token to use as streaming currency
+    - `isEntity`: indicates whether the stream exists or not
+
+```solidity
+struct Stream {
+    uint256 deposit;
+    uint256 ratePerSecond;
+    uint256 remainingBalance;
+    uint256 startTime;
+    uint256 stopTime;
     address recipient;
+    address sender;
     address tokenAddress;
-    uint256 balance;
-    Timeframe timeframe;
-    Rate rate;
-  }
-```
-
-- `timeframe`
-    - `start`: the starting block number of the stream
-    - `stop`: the stopping block number of the stream
-
-```solidity
-struct Timeframe {
-    uint256 start;
-    uint256 stop;
-}
-```
-
-- `rate`
-    - `payment`: how much money moves from `sender` to `recipient`
-    - `interval`: how often `payment` moves from `sender` to `recipient`
-
-```solidity
-struct Rate {
-  uint256 payment;
-  uint256 interval;
+    bool isEntity;
 }
 ```
 
 ---
 
 ### Methods
 
-#### balanceOf
-
-Returns available funds for the given stream id and address.
-
-```solidity
-function balanceOf(uint256 _streamId, address _addr)
-```
-
 #### getStream
 
-Returns the full stream data, if the id points to a valid stream.
-
-```solidity
-function getStream(uint256 _streamId) returns (address sender, address recipient, address tokenAddress, uint256 balance, uint256 startBlock, uint256 stopBlock, uint256 payment, uint256 interval)
-```
-
-#### create
-
-Creates a new stream between `msg.sender` and `_recipient`.
-
-MUST allow senders to create multiple streams in parallel. SHOULD not accept Ether and only use ERC20-compatible tokens.
-
-**Triggers Event**: [LogCreate](#logcreate)
+Returns the stream object with all its properties.
 
 ```solidity
-function create(address _recipient, address _tokenAddress, uint256 _startBlock, uint256 _stopBlock, uint256 _payment, uint256 _interval)
+function getStream(uint256 streamId) view returns (address sender, address recipient, uint256 deposit, address token, uint256 startTime, uint256 stopTime, uint256 remainingBalance, uint256 ratePerSecond)
```

Comment:
If you are using Solidity syntax here, then I suggest to perhaps consider formatting it according to the style guide for readability, but at least add a semicolon to the end, so it could be copy pasted into an interface.

Furthermore some ERCs chose to describe it as a Solidity interface.